demo/index.html

```diff
@@ -7,6 +7,8 @@
 
   <title>vaadin-text-field</title>
 
+  <!-- Safari has severe rendering issues with native Shadow DOM -->
```

Comment:
https://bugs.webkit.org/show_bug.cgi?id=170762

Already fixed in WebKit. Not yet shipped, though.